include/mips-emulator/executor.hpp

```diff
@@ -339,6 +339,25 @@ namespace mips_emulator {
 
             const ROp op = static_cast<ROp>(instr.special3_rtype.op);
             switch (op) {
+                case ROp::e_bitswap: {
```

Comment:
Why don't you use a for loop?
```c++
uint32_t result = 0;
for (int i = 0; i < 32; i++)
    result |= ((rt.u >> i) & 1) << (31- i);

reg_file.set_unsigned(instr.special3_rtype.rd, result);
```

or even faster:
https://stackoverflow.com/a/9144870

```c++
uint32_t reverse(uint32_t x)
{
    x = ((x >> 1) & 0x55555555u) | ((x & 0x55555555u) << 1);
    x = ((x >> 2) & 0x33333333u) | ((x & 0x33333333u) << 2);
    x = ((x >> 4) & 0x0f0f0f0fu) | ((x & 0x0f0f0f0fu) << 4);
    x = ((x >> 8) & 0x00ff00ffu) | ((x & 0x00ff00ffu) << 8);
    x = ((x >> 16) & 0xffffu) | ((x & 0xffffu) << 16);
    return x;
}
```